Sources/VSM/Documentation.docc/Reference/ViewStateExtensions.md

```diff
@@ -1,12 +1,232 @@
+
+The following simplified example highlights one of these situations:
+
+```swift
+struct EditUserProfileView: View, ViewStateRendering {
+    ...
+    var body: some View {
+        switch state.editingState {
+        case .editing:
+            TextField("Username", $username)
+                .disabled(false)
+        case .saving:
+            TextField("Username", $username)
+                .disabled(true)
+        }
+    }
+}
+```
+
+While you may think these two view's identities are the same, they are in fact not. The SwiftUI framework sees these as two separate text fields nested within a switch view: `SwitchView<TextField, TextField>`.
+
+The problem we have here is that our view state is an enum and our view code will become much more complicated and verbose if we try to convert the enum to a boolean like so:
+
+```swift
+struct EditUserProfileView: View, ViewStateRendering {
+    ...
+    var body: some View {
+        let isSaving: Bool
+        switch state.editingState {
+        case .editing:
+            isSaving = false
+        case .saving:
+            isSaving = true
+        }
+        TextField("Username", $username)
+            .disabled(isSaving)
+        ...
+    }
+}
+```
+
+To prevent our view's code from exploding into data manipulation code, we can instead opt for a much cleaner approach. By extending the view state, we can add a computed property for getting at the information we need in the format that is most convenient.
+
+```swift
+extension EditUserProfileViewState {
+    var isSaving: Bool {
+        switch editingState {
+        case .editing:
+            return false
+        case .saving:
+            return true
+        }
+    }
+}
+```
+
+Now that we have this extension, our view code becomes much cleaner.
+
+```swift
+struct EditUserProfileView: View, ViewStateRendering {
+    ...
+    var body: some View {
+        TextField("Username", $username)
+            .disabled(state.isSaving)
+        ...
+    }
+}
+```
+
+SwiftUI now sees this TextField as a single view with a disabled modifier. Any transitions, animations, or memory optimizations are now guaranteed to work for this view. These view state extensions are very handy for transforming the Feature Shape into something more digestible for the UI.
+
+## Binding to SwiftUI Views
+
+When working with the SwiftUI framework, you will often come across views or view modifiers that require a `Binding<T>` of some kind. For example, you may have an error view that you want to display if the user gets into an error state. Without view state extensions, your view will be cluttered with code that translates the view state enum into a `Binding<Bool>` and a confusing `.sheet()` view modifier implementation.
+
+```swift
+struct EditUserProfileView: View, ViewStateRendering {
+    ...
+    var hasError: Binding<Bool>
+
+    init() {
+        _hasError = Binding<Bool> {
+            get: {
```

Comment:
Is the trailing colon correct here (and for set, below)?

Reply:
It is not! good catch!